### PR TITLE
Add props for loading spinner text

### PR DESCRIPTION
**Pull request by alexabatino**
## Description of the change

Give the Loading component props so that the text can be customized for the spinner

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

> Closes [#XXXX]

### Development

These boxes should be checked by the submitter prior to merging:

- [ ] Manual testing against realistic data has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: packages/design-system/src/components/Loading/Loading.types.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+// =============================================================================
+
+export interface LoadingProps extends React.HTMLAttributes<any> {
```
Comment (macfarlandian) [inline packages/design-system/src/components/Loading/Loading.types.ts:18]: lint warning here ... but also you aren't using any of the html props anyway so I don't think you even need to extend this

**Review by macfarlandian — COMMENTED** [review]
i would also just include a package version bump and changelog update in this PR, so you can publish it right after merging

File: packages/design-system/src/components/Loading/Loading.types.ts (hunk 1 above)
Comment (alexabatino) [reply packages/design-system/src/components/Loading/Loading.types.ts:18]: Hm, I added this to address an error in `dashboards` when I passed the prop [here](https://github.com/Recidiviz/recidiviz-dashboards/pull/2980/files#diff-01161f71372a489d1ac154b7536f7eadd1d000630f0531880101b57310a25423R156). Error said something like `Type '{ showMessage: boolean; }' is not assignable to type 'IntrinsicAttributes & Props'`

Comment (macfarlandian) [reply packages/design-system/src/components/Loading/Loading.types.ts]: I think just passing the props type to `React.FC` should address that, without the need for the HTMLAttributes interface

Comment (ohaibbq) [reply packages/design-system/src/components/Loading/Loading.types.ts]: TIL! Would we need to specify a HTML attributes super if we wanted to be able props for a specific tag type i.e. `<button/>`, or does FC cover all possible tag props?

Comment (alexabatino) [reply packages/design-system/src/components/Loading/Loading.types.ts]: Retried it without the extends and it worked! removed

Comment (macfarlandian) [reply packages/design-system/src/components/Loading/Loading.types.ts]: @ohaibbq not sure I am understanding your question right but we have done stuff like that before, e.g. [here](https://github.com/Recidiviz/web-libraries/blob/main/packages/design-system/src/components/Button/Button.types.ts#L24) ... FC doesn't assume anything HTML-specific out of the box

Comment (ohaibbq) [reply packages/design-system/src/components/Loading/Loading.types.ts]: Ah, gotcha, I had interpreted your original comment as if `React.FC` _**did**_  include HTML-specific props out of the box. That specific button implementation is what I had in mind.

**Review by macfarlandian — APPROVED** [review]

**Pull request merged**